tests/range/range_api.cpp

```diff
@@ -212,6 +212,20 @@ class test_range {
   }
 };
 
+// This tests is only here to check if one can submit a large kernel.
+// We don't use an `empty kernel` to avoid smart compiler
+// optimizing the submision away
+// This test:
+//    - Doesn't check that all the work-item have be submited
+//    - It's technicaly UB (concurent write)
+void test_launch_kernel_1d_range(size_t N, sycl::queue q) {
+  int *a = sycl::malloc_shared<int>(q, 1);
+  a[0] = 0;
+  q.parallel_for(sycl::range<1>(N), [=](auto i) { a[0] = 1; }).wait_and_throw();
```

Comment:
Please, "name" the kernel. ComputeCPP doesn't support unnamed kernels.

Reply:
Is this not now part of the spec now? :( Will do! I guess I need to use template to avoid some double-definition errors in case of inlining.

Reply:
I think we should also check that device supports that number of work-items (i.e. N <= info::device::max_work_item_sizes<1>).

Reply:
> Is this not now part of the spec now? :( Will do! I guess I need to use template to avoid some double-definition errors in case of inlining.

You are right. Technically, it's required unless the implementation supports only reduced feature-set. You can just disable the testing with ComputeCPP then.

Reply:
This is the trick `max_work_item_sizes` is only for the work items inside the work group of the nd_range.  It's unrelated to the absolute number of work items supported inside `range` used in `parralel_for`. Those are only bounded by `size_t` AFAIK.